pkg/k8s/namespaces/crud.go

```diff
@@ -21,3 +27,100 @@ func IsOktetoNamespace(ns string, c *kubernetes.Clientset) bool {
 	}
 	return n.Labels[OktetoLabel] == "true"
 }
+
+//CheckAvailableResources checks if therre are available resources for activating the dev env
+func CheckAvailableResources(dev *model.Dev, create bool, d *appsv1.Deployment, c *kubernetes.Clientset) error {
```

Comment:
is it possible to split the part where we call kubernetes api from the rest? that way we can add unit tests to the limit calculation and validate the different scenarios.

Reply:
it would also make it easier to follow if we had at least one function where we collect the  information from the namespace, another where we calculate the needs of the `up` and a third one that makes the decision.